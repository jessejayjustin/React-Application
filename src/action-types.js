export const FETCH_CHARITIES = 'FETCH_CHARITIES';
export const UPDATE_TOTAL_DONATE = 'UPDATE_TOTAL_DONATE';
export const UPDATE_MESSAGE = 'UPDATE_MESSAGE';

