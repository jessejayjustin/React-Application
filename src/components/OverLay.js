import React, { Component } from "react";
import styled from 'styled-components';

class OverLay extends Component {
  constructor() {
    super();
    this.state = {
      selectedItem: [],
      selectedAmount: 10
    };
  }

  handlePay(id, amount, currency) {
    this.setState(
      {
        selectedItem: {
          id: id,
          amount: amount,
          currency: currency
        }
      },
      function() {
        this.props.handlePay(this.state.selectedItem);
      }
    );
  }

  selectedAmount(amount) {
    this.setState({ selectedAmount: amount })
  }

  render() {
  
    const {payments, item, hideOverlay} = this.props;

    const overLayStyle = {
      backgroundColor: 'white',
      transition: '0.5s',
      zIndex: '1',
      width: '530px',
      height: '320px',
      opacity: '0.9',
      position: 'absolute',
      left: '0px',
      top: '0px'
    }

    const inputStyle = {
      display: 'inline-block',
      float: 'left',
      width: '12%',
      position: 'relative',
      left: '115px',
      top: '135px',
    }

    const textStyle = {
      fontSize: '18px',
      color: 'rgb(96,96,96)',
      fontFamily: 'Trebuchet MS, Helvetica, sans-serif'
    }

    const payBtn = {
      position: 'absolute',
      top: '190px',
      left: '245px',
      padding: '5px 9px',
      backgroundColor: 'rgb(255,255,255)',
      color: 'rgb(30,144,255)',
      fontFamily: 'system-ui',
      border: '2px solid rgb(30,144,255)',
      borderRadius: '3px',
      fontSize: '15px',
      fontWeight: 'bold',
      cursor: 'pointer'
    };

    const paymentText = {
      position: 'absolute',
      top: '70px',
      left: '119px',
      fontSize: '19px',
      color: 'rgb(96,96,96)',
      textAlign: 'center',
      fontFamily: 'Trebuchet MS, Helvetica, sans-serif'
    };

    const closeBtn = {
      position: 'absolute',
      right: '20px',
      top: '22px',
      zIndex: '1',
      cursor: 'pointer',
      fontSize: '22px',
      fontWeight: 'bold'
    };

    return (
      <div style={overLayStyle}>
      <span style={closeBtn} onClick={hideOverlay}>&#215;</span>
      <p style={paymentText}>Select the amount to donate (EUR)</p>
      {payments.map((amount, index) => {
          return (
            <div style={inputStyle} key={index} className="radio">
            <label>
              <input
               name="payment"
               type="radio"
               value={amount}
               onClick={() => {this.selectedAmount(amount)}}
              /><span></span><span style={textStyle}>{amount}</span>
            </label>
            </div>
          )
      })}
      <button 
      style={payBtn}
      onClick={this.handlePay.bind(
      this,
      item.id,
      this.state.selectedAmount,
      item.currency
      )}
      >Pay
      </button>
      </div>
    );
  }
}
 
export default OverLay;
