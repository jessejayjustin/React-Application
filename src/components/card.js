import React, { useEffect, useRef, useState } from 'react';
import Input from './input';

const card = ({id, name, image, currency, currIndex, handlePayment, handleOverlay}) => {
  
  const [amt, setAmt] = useState(10);
  const [payments, setPayments] = useState([10,20,50,100,500]);

  const refs = useRef([React.createRef(), React.createRef()]);

  const overlayRef = useRef();
  const openOverlayRef = useRef();
  const closeOverlayRef = useRef();

  useEffect(() => {

    const handleOverlayOpen = () => {
      overlayRef.current.style.width = '100%';
      refs.current[0].current.checked = "true";
    }
    const handleOverlayClose = () => overlayRef.current.style.width = '0';

    openOverlayRef.current.addEventListener('click', handleOverlayOpen);
    closeOverlayRef.current.addEventListener('click', handleOverlayClose);

    return () => {
      openOverlayRef.current.removeEventListener('click', handleOverlayOpen);
      closeOverlayRef.current.removeEventListener('click', handleOverlayClose);
    }

  }, [])

  return (
    <div className="col-lg-5 col-md-9 col-sm-11 column">
      <div className="card">
        <div ref={overlayRef} className="overlay">
          <a ref={closeOverlayRef} className="float-right mr-3 mt-3 close-btn">x</a>
          <p className="title text-center">Select the amount to donate</p>
          <div className="btn-group d-flex justify-content-center">
            {
              payments.map((amount, index) => {
                return (
                  <Input 
                   className="inpt"
                   key={index}
                   Id={id}
                   index={index}
                   type="radio"
                   value={amount} 
                   currIndex={currIndex}
                   ref={refs.current[index]}
                   id={currIndex===id ? index : ""} 
                   name="customRadioInline1" 
                   onClick={() => setAmt(amount)}
                  />
                )
              })
            }
          </div>
          <button 
           className="btn btn-sm btn-outline-primary mt-5 d-flex mx-auto"
           disabled={currIndex===id ? false : true}
           onClick={handlePayment.bind(
            this,
            id,
            amt,
            currency
           )}
           >Pay
           </button>
        </div>
        <img className="img-fluid" src={require(`../../public/images/${image}`)} />
        <div className="card-body">
          <h5 className="card-title">{name}<a href="#" className="float-right"></a></h5>
          <button ref={openOverlayRef} className="btn btn-sm btn-outline-primary float-right" onClick={handleOverlay}>Donate</button>
        </div>
      </div>
    </div>
  )
}

export default card;





