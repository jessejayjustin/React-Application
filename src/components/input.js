import React from 'react';

function Input({Id, index, type, value, currIndex, id, name, onClick}, ref) {
  return (
    <div key={index} className="custom-control custom-radio custom-control-inline">
      <input 
        type={type} 
        ref={ref}
        id={id} 
        name={name}
        className="custom-control-input"
        value={value} 
        onClick={onClick}
        />
      <label className="custom-control-label" htmlFor={currIndex===Id ? index : ""}>{value}</label>
    </div>  
  )
}

const forwardRef = React.forwardRef(Input);

export default forwardRef;