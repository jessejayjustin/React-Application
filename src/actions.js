import { FETCH_CHARITIES, UPDATE_TOTAL_DONATE, UPDATE_MESSAGE } from "./action-types";
import { summaryDonations } from './helpers';

export function fetchCharities() {
    return dispatch => {
      return fetch('http://localhost:3001/charities')
        .then(function(resp) { return resp.json(); })
        .then(function(data) {
            dispatch({type: FETCH_CHARITIES, payload: data});
        }).catch(response => {
          console.log(response)
        });
    };
};

export function fetchPayments() {
    return dispatch => {
      return fetch('http://localhost:3001/payments')
        .then(function(resp) { return resp.json(); })
        .then(function(data) {
        	//console.log(data)
            dispatch({
            	type: UPDATE_TOTAL_DONATE, 
            	payload: summaryDonations(data.map((item) => (item.amount)))
            });
        }).catch(response => {
          console.log(response)
        });
    };
};

export function handleAddPayment({id, amount, currency}) {
    return dispatch => {
       return fetch('http://localhost:3001/payments', {
        method: 'POST',
        body: `{ "charitiesId": ${id}, "amount": ${amount}, "currency": "${currency}" }`,
        headers: { /* POST request should include a Content-Type: application/json header to use the JSON in the request body. Otherwise it will result in a 200 OK but without changes being made to the data. */
          Accept: "application/json",
          "Content-Type": "application/json"
        },
      }).then((response) => {
        response.json().then(data =>{
          dispatch({type: UPDATE_TOTAL_DONATE, payload: data.amount});
          dispatch({
            type: UPDATE_MESSAGE,
            payload: `Thanks for donate ${data.amount}`,
          });
          setTimeout(function() {
            dispatch({
              type: UPDATE_MESSAGE,
              message: ``,
            });
          }, 2000);
        })
      }).catch(response => {
          console.log(response)
      });
    };
};