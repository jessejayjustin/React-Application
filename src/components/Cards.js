import React, { Component } from "react";
import styled from 'styled-components';
import OverLay from "./OverLay";

const Container = styled.div`
  background-color: white;
  width: 530px;
  height: 320px;
  position: relative;
  left: 115px;
  margin: 20px 20px;
  float: left;
  border-radius: 5px;
  -moz-box-shadow: 0px 3px 11px rgb(211,211,211);
  -webkit-box-shadow: 0px 3px 11px rgb(211,211,211);
  box-shadow: 0px 3px 15px rgb(211,211,211);
`;

export class Cards extends Component {
  constructor() {
    super();

    this.state = {
      selectedAmount: 10,
      active: null,
      activate: null
    };
    this.myVisibility = this.myVisibility.bind(this);   // Bind returns a new function, allowing you to pass in a this array and any number of arguments.
    this.active = this.active.bind(this);
    this.disable = this.disable.bind(this);
    this.btnDisabled = this.btnDisabled.bind(this);
    this.hideOverlay = this.hideOverlay.bind(this);
  }

  active(index) {
    if (this.state.active === index) {
      this.setState({active : null})
    } else {
      this.setState({active : index})
    }
  }

  myVisibility(index) {
    if (this.state.active === index) {
      return "visible";
    }
    return "hidden";
  }

  disable(item) {
    if (this.state.activate === item) {
      this.setState({activate : null})
    } else {
      this.setState({activate : item})
    }
  }

  btnDisabled(item) {
    if (this.state.activate === item) {
      return true;
    }
    return false;
  }

  hideOverlay() {
    //console.log("button clicked")
    this.setState({active : null})
    this.setState({activate : null})
  }

  render() {

    const imageStyle = {
      position: 'absolute',
      top: '0px',
      right: '0px',
      zIndex: '1',
      maxHeight:'100%',
      width: '530px',
      height: '245px',
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'center'
    };

    const btnStyle = {
      position: 'absolute',
      bottom: '23px',
      right: '25px',
      padding: '4px 9px',
      backgroundColor: 'rgb(255,255,255)',
      color: 'rgb(30,144,255)',
      fontFamily: 'system-ui',
      border: '2px solid rgb(30,144,255)',
      borderRadius: '3px',
      fontSize: '15px',
      fontWeight: 'bold',
      cursor: 'pointer'
    };

    const itemName = {
      position: 'absolute',
      bottom: '11px',
      left: '22px',
      fontSize: '17px',
      color: 'rgb(96,96,96)',
      fontFamily: 'Trebuchet MS, Helvetica, sans-serif'
    };

    const {charities, payments, handlePay} = this.props;
   
    return (
      <div className="app">
      {charities.map((item, index) => {
          return (
            <Container key={index} id="container">
            <img style={imageStyle} src={require(`../../public/images/${item.image}`)} alt="The image" className="img-responsive" />
            <p style={itemName}>{item.name}</p>
           
            <button style={btnStyle} onClick={() => { this.active(index); this.disable(index) }} disabled={this.btnDisabled(index)}>Donate</button> {/* onClick handler, invoking multiple functions */}
            
            <div style={{visibility:this.myVisibility(index)}}> {/* Using identifiers on map elements i can now activate/deactivate them all every time. By referencing each of them in a different way, then i can set the visibility property individually. */}
            <OverLay 
              payments={payments}
              item={item}
              handlePay={handlePay}
              hideOverlay={this.hideOverlay}
            />
            </div>
            </Container>
          )
      })}
      </div>
    );
  }
}

export default Cards;




