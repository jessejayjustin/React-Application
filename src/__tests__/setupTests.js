import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk'
import { mount, shallow, render } from 'enzyme';
import nock from 'nock';
import SmartApp, { App } from '../App';
import { Overlay } from '../components/Overlay';
import * as actions from '../actions';
import * as types from '../action-types';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
configure({ adapter: new Adapter() });

describe('Components', () => {
  let wrapper, container, containerProp, childContainer, childContainerProps;

  const props = {
    fetchCharities: jest.fn(),
    fetchPayments: jest.fn() 
  }

  beforeEach(() => {
    wrapper = shallow(<App {...props} />);
    container = wrapper.find("div");
    containerProp = container.props();
    childContainer = wrapper.find('Cards');
    childContainerProps = childContainer.props();
  });

  it('it should render the <App/> component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a <div>", () => {
    expect(container).toHaveLength(1);
  });

  it("should have donate as prop", () => {
    expect(containerProp.donate).toEqual();
  });

  it("should have message as prop", () => {
    expect(containerProp.message).toEqual();
  });

  it("should have a <div> with properly className prop", () => {
    expect(containerProp.className).toEqual("app");
  });

  it("should have a <ChildContainer>", () => {
    expect(childContainer).toHaveLength(1);
  });

  it("should have payments as prop", () => {
    expect(childContainerProps.payments).toEqual([10, 20, 50, 100, 500]);
  });

  it("should have charities as prop", () => {
    expect(childContainerProps.charities).toEqual();
  });

  it("should have handleAddPayment as prop", () => {
    expect(childContainerProps.handleAddPayment).toEqual();
  });
});

describe('>>><SmartApp/> --- REACT-REDUX (wrapping in <Provider>)',()=>{
    const initialState = {
      donate: 0,
      charities: [],
      message: ''
    }
    const mockStore = configureMockStore()
    let store, wrapper;

    beforeEach(()=>{
        store = mockStore(initialState)
        wrapper = shallow(<Provider store={store}><SmartApp/></Provider>)
    });

    it('+++ render the <SmartApp/> component', () => {
       expect(wrapper.find(SmartApp).length).toEqual(1)
    });
});

describe('action', () => {
  it('should create an action to fetchCharities', () => {
    expect(actions.fetchCharities({})).toBeDefined;
  })
});

describe('action', () => {
  it('should create an action to fetchPayments', () => {
    expect(actions.fetchPayments({})).toBeDefined;
  })
});

describe('action', () => {
  it('should create an action to handleAddPayment', () => {
    expect(actions.handleAddPayment({})).toBeDefined;
  })
});

/* Asynchronous actions, HTTP request to a server, npm run server */
const middlewares = [thunk] // add your middlewares like `redux-thunk`
const mockStore = configureMockStore(middlewares)

function fetchCharitiesSuccess() {
  return {
    type: 'FETCH_CHARITIES'
  }
}

function fetchCharities () {
  return dispatch => {
    return fetch('http://localhost:3001/charities') // Some async action with promise
      .then(() => dispatch(fetchCharitiesSuccess()))
  };
}

it('should execute fetch charities data', () => {
  const store = mockStore({})

  // Return the promise
  return store.dispatch(fetchCharities())
    .then(() => {
      const actions = store.getActions()
      expect(actions[0]).toEqual(fetchCharitiesSuccess())
    })
})

function fetchPaymentsSuccess() {
  return {
    type: 'UPDATE_TOTAL_DONATE'
  }
}

function fetchPayments () {
  return dispatch => {
    return fetch('http://localhost:3001/payments') // Some async action with promise
      .then(() => dispatch(fetchPaymentsSuccess()))
  };
}

it('should execute fetch payments data', () => {
  const store = mockStore({})

  // Return the promise
  return store.dispatch(fetchPayments())
    .then(() => {
      const actions = store.getActions()
      expect(actions[0]).toEqual(fetchPaymentsSuccess())
    })
})

function handleAddPaymentSuccess() {
  return {
    type: 'UPDATE_TOTAL_DONATE'
  }
}

function handleAddPayment () {
  return dispatch => {
    return fetch('http://localhost:3001/payments') // Some async action with promise
      .then(() => dispatch(handleAddPaymentSuccess()))
  };
}

it('should execute post payments data', () => {
  const store = mockStore({})

  // Return the promise
  return store.dispatch(handleAddPayment())
    .then(() => {
      const actions = store.getActions()
      expect(actions[0]).toEqual(handleAddPaymentSuccess())
    })
})
/* Asynchronous actions, HTTP request to a server, npm run server */

const ROOT_URL = 'http://localhost:3001';

describe('actions', () => {

    beforeEach(() => {
        nock.disableNetConnect();
    });

    afterEach(() => {
        nock.cleanAll();
        nock.enableNetConnect();
    });

  describe('charities', () => {

    it('has the correct type', () => {
      var scope = nock(ROOT_URL).get('/charities').reply(200,{ charities: [{ id: 1, name: 'Baan Kru Noi', image: 'baan-kru-noi.jpg', currency: 'THB' }] });
      const store = mockStore({ charities: [] });

      store.dispatch(fetchCharities()).then(() => {
      const actions = store.getStore()
      expect(actions.charities).toEqual([{ id: 1, name: 'Baan Kru Noi', image: 'baan-kru-noi.jpg', currency: 'THB' }]);
    })

    });
  });
});

describe('actions', () => {

    beforeEach(() => {
        nock.disableNetConnect();
    });

    afterEach(() => {
        nock.cleanAll();
        nock.enableNetConnect();
    });

  describe('charities', () => {

    it('has the correct type', () => {
      var scope = nock(ROOT_URL).get('/payments').reply(200,{ payments: [{ charitiesId: 5, amount: 500, currency: 'THB', id: 1 }] });
      const store = mockStore({ payments: [] });

      store.dispatch(fetchPayments()).then(() => {
      const actions = store.getStore()
      expect(actions.payments).toEqual([{ charitiesId: 5, amount: 500, currency: 'THB', id: 1 }]);
    })

    });
  });
});





