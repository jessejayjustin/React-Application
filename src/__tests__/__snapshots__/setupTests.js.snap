// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components it should render the <App/> component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    fetchCharities={[Function]}
    fetchPayments={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          style={
            Object {
              "color": "rgb(96,96,96)",
              "fontFamily": "Trebuchet MS, Helvetica, sans-serif",
              "fontSize": "39px",
              "fontWeight": "bold",
              "lineHeight": ".5em",
              "marginTop": "1em",
              "textAlign": "center",
            }
          }
        >
          Omise Tamboon React
        </h1>,
        <p
          style={
            Object {
              "color": "red",
              "fontSize": "17px",
              "fontWeight": "bold",
              "lineHeight": "0",
              "margin": "2em 0",
              "textAlign": "center",
            }
          }
        >
          All donations: 
        </p>,
        <p
          style={
            Object {
              "color": "red",
              "fontSize": "16px",
              "fontWeight": "bold",
              "lineHeight": "1em 0",
              "margin": "1em 0",
              "textAlign": "center",
            }
          }
        />,
        <Cards
          charities={undefined}
          handlePay={[Function]}
          payments={
            Array [
              10,
              20,
              50,
              100,
              500,
            ]
          }
        />,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Omise Tamboon React",
          "style": Object {
            "color": "rgb(96,96,96)",
            "fontFamily": "Trebuchet MS, Helvetica, sans-serif",
            "fontSize": "39px",
            "fontWeight": "bold",
            "lineHeight": ".5em",
            "marginTop": "1em",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": "Omise Tamboon React",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "All donations: ",
            undefined,
          ],
          "style": Object {
            "color": "red",
            "fontSize": "17px",
            "fontWeight": "bold",
            "lineHeight": "0",
            "margin": "2em 0",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          "All donations: ",
          undefined,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "style": Object {
            "color": "red",
            "fontSize": "16px",
            "fontWeight": "bold",
            "lineHeight": "1em 0",
            "margin": "1em 0",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "charities": undefined,
          "handlePay": [Function],
          "payments": Array [
            10,
            20,
            50,
            100,
            500,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
              Object {
                "color": "rgb(96,96,96)",
                "fontFamily": "Trebuchet MS, Helvetica, sans-serif",
                "fontSize": "39px",
                "fontWeight": "bold",
                "lineHeight": ".5em",
                "marginTop": "1em",
                "textAlign": "center",
              }
            }
          >
            Omise Tamboon React
          </h1>,
          <p
            style={
              Object {
                "color": "red",
                "fontSize": "17px",
                "fontWeight": "bold",
                "lineHeight": "0",
                "margin": "2em 0",
                "textAlign": "center",
              }
            }
          >
            All donations: 
          </p>,
          <p
            style={
              Object {
                "color": "red",
                "fontSize": "16px",
                "fontWeight": "bold",
                "lineHeight": "1em 0",
                "margin": "1em 0",
                "textAlign": "center",
              }
            }
          />,
          <Cards
            charities={undefined}
            handlePay={[Function]}
            payments={
              Array [
                10,
                20,
                50,
                100,
                500,
              ]
            }
          />,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Omise Tamboon React",
            "style": Object {
              "color": "rgb(96,96,96)",
              "fontFamily": "Trebuchet MS, Helvetica, sans-serif",
              "fontSize": "39px",
              "fontWeight": "bold",
              "lineHeight": ".5em",
              "marginTop": "1em",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Omise Tamboon React",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "All donations: ",
              undefined,
            ],
            "style": Object {
              "color": "red",
              "fontSize": "17px",
              "fontWeight": "bold",
              "lineHeight": "0",
              "margin": "2em 0",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "All donations: ",
            undefined,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "style": Object {
              "color": "red",
              "fontSize": "16px",
              "fontWeight": "bold",
              "lineHeight": "1em 0",
              "margin": "1em 0",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "charities": undefined,
            "handlePay": [Function],
            "payments": Array [
              10,
              20,
              50,
              100,
              500,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
